### PR TITLE
Kraken: base_*_date_time calculation for arrivals/departures APIs

### PR DESCRIPTION
**Pull request by azime**
see: https://github.com/CanalTP/navitia/issues/1948
jira: http://jira.canaltp.fr/browse/NAVITIAII-2207

### CODE REVIEW

File: source/jormungandr/tests/departure_board_tests.py [hunk 1]
```diff
@@ -779,3 +779,97 @@ def test_departure_dt_base_schedule(self):
...
+                                     "&data_freshness=base_schedule")
+
+        departures = response["departures"]
+
+        is_valid_departures(departures)
+        assert len(departures) == 1
+        assert departures[0]["route"]["id"] == 'M:6'
+        assert departures[0]["stop_point"]["id"] == 'S11'
+        assert departures[0]["stop_date_time"]["arrival_date_time"] == '20160101T113000'
+        assert departures[0]["stop_date_time"]["departure_date_time"] == '20160101T113500'
```
Comment (TeXitoi) [inline source/jormungandr/tests/departure_board_tests.py]: Why you don't check the `base_` variants ?

**Review by TeXitoi — COMMENTED** [review]
Lack tests arround "RT vj begin the day after"

File: source/jormungandr/tests/departure_board_tests.py (hunk 1 above)
Comment (azime) [reply source/jormungandr/tests/departure_board_tests.py]: ok

File: source/jormungandr/tests/departure_board_tests.py [hunk 2]
```diff
@@ -779,3 +779,241 @@ def test_departure_dt_base_schedule(self):
...
+        is_valid_departures(arrivals)
+        assert len(arrivals) == 1
+        assert arrivals[0]["route"]["id"] == 'M:6'
+        assert arrivals[0]["stop_point"]["id"] == 'S11'
+        assert arrivals[0]["stop_date_time"]["arrival_date_time"] == '20160101T113000'
+        assert arrivals[0]["stop_date_time"]["departure_date_time"] == '20160101T113500'
+
+    def test_arrival_schedule_arrivals_date_time_not_eq_arrival_date_time_realtime(self):
+        """
+        test a arrival, with a date time
```
Comment (antoine-de) [inline source/jormungandr/tests/departure_board_tests.py]: I find the tests a bit diffucult to read, the comment is a bit cryptic :wink:

File: source/tests/utils_test.h [hunk 3]
```diff
@@ -32,12 +32,14 @@ inline uint64_t to_posix_timestamp(const std::string& str) {
...
 inline transit_realtime::TripUpdate
 make_delay_message(const std::string& vj_uri,
-        const std::string& date,
+        const std::string& start_date,
         const std::vector<std::tuple<std::string, int, int>>& delayed_time_stops) {
     transit_realtime::TripUpdate trip_update;
     auto trip = trip_update.mutable_trip();
     trip->set_trip_id(vj_uri);
-    trip->set_start_date(date);
+    // start_date: see
```
Comment (antoine-de) [inline source/tests/utils_test.h]: to ease reading, maybe you can write:

```
// start_date is used to disambiguate trips that are very late, cf:
// https://github.com/CanalTP/chaos-proto/blob/master/gtfs-realtime.proto#L459
```

**Review by TeXitoi — APPROVED** [review]

File: source/tests/utils_test.h (hunk 3 above)
Comment (azime) [reply source/tests/utils_test.h]: Done

File: source/jormungandr/tests/departure_board_tests.py (hunk 2 above)
Comment (azime) [reply source/jormungandr/tests/departure_board_tests.py]: Done

**Comment by azime** [comment]
Jenkins, test this please

**Pull request merged**